base/expr.jl

```diff
@@ -366,9 +366,17 @@ function remove_linenums!(ex::Expr)
             isa(x, LineNumberNode) && return false
             return true
         end
+    elseif ex.head === :macrocall
+        # Replace line information embedded into macro calls with `nothing`
+        # Removing the argument entirely invalidates the expression
+        map!(ex.args, ex.args) do arg
+            arg isa LineNumberNode ? LineNumberNode(1, :none) : remove_linenums!(arg)
```

Comment:
Maybe one could use `LineNumberNode(0, nothing)` as an replacement, to be consistent with the behaviour of `@__LINE__` and `@__FILE__`?